teps/0057-tekton-catalog-pipeline-organization.md

```diff
@@ -0,0 +1,477 @@
+./{resource-type}/{resource-name}/{version}/install-script.sh
+
+# Tasks directory which refers to the task used in the pipeline
+./{resource-type}/{resource-name}/{version}/tasks/..
+
+# Optional: Test working samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/tests/…
+# Optional: Addition samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/samples/…
+```
+
+For example:
+
+```bash=
+
+./pipeline/                     👈 The kind of the resource
+
+    /p1                         👈 Definition file must have same name
+       /OWNERS                  👈 owners of this resource
+       /0.1
+         /README.md
+         /install-script.sh     👈 To install all resources related to pipeline
+         /p1.yaml               👈 The file name should match the resource name
+         /samples/
+         /tasks                 👈 Tasks which pipelines refers
+         /support
+       /0.2/...
+
+    /p2
+       /OWNERS
+       /README.md
+       /0.1
+         /README.md
+         /p2.yaml
+         /samples/
+         /tasks
+```
+
+#### Fields added to the yaml file
+
+```yaml=
+metadata:
+  name: <pipeline-name>                           👈 MUST: Name of file and pipeline name should be the same
+  labels:
+    app.kubernetes.io/version: "0.1"              👈 MUST: version of the resource
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"     👈 MUST: version of pipeline
+    tekton.dev/category: ""                      👈 MUST: Category of a resource
+    tekton.dev/tags: <tags>                       👈 Optional: Comma separated list of tags
+    tekton.dev/displayName: ""                    👈 Optional: Display name of the pipeline
+  spec:
+    description: >-
+      The Tekton Pipelines project provides k8s-style
+      resources for declaring CI/CD-style pipelines.             👈 #     MUST:  One line Summary of the task
+
+      One Pipeline can be used to deploy to any k8s cluster.
+      The Tasks which make up a Pipeline can easily be run
+      in isolation.Resources such as git repos can easily be
+      swapped between runs                                       👈 # Optional: Description
+```
+
+#### Scenarios of task references in a pipeline
+
+Pipelines could include any combination of tasks from the Tekton catalog, other catalog and tasks included in the tasks folder of the pipeline definition.
+
+Here are some sample scenarios:
+
+- Pipeline can have tasks which are part of tekton catalog and they can be referred via annotations
+
+- Pipeline may have tasks out of which some tasks might come from the catalog and some
+  may be specific to that pipeline which have to be added to the `/tasks` dir
+
+- Pipeline can have tasks which are just specific to that pipeline and that
+  tasks can be added in the `/tasks` directory
+
+- Pipelines can have tasks which can be a part of other catalogs
```

Comment:
> To be honest I'm not sure if we want pipeline in the Tekton catalog that refer to tasks in other catalogs, at least not in the "official" and "verified" tiers?

This was my reaction as well.  I don't think the OSS catalog should allow references to external Tasks. Is there a reason to support that?